numpy/ma/core.py

```diff
@@ -3802,7 +3802,10 @@ def __iadd__(self, other):
         else:
             if m is not nomask:
                 self._mask += m
-        ndarray.__iadd__(self._data, np.where(self._mask, 0, getdata(other)))
+        ndarray.__iadd__(
```

Comment:
That's right. I forgot about this, `_data` is defined as readonly [numpy/ma/core.py#L3365](https://github.com/numpy/numpy/blob/4bfcf4801ceedd0563e3648154fc49985729c442/numpy/ma/core.py#L3365). It still seems a bit weird. Maybe, it is determining that `__iadd__` doesn't exist for the property and goes with `__add__` followed by assignment, which fails as there is no `fset` specified.